src/js/jsx/shared/Label.jsx

```diff
@@ -25,10 +25,47 @@ define(function (require, exports, module) {
     "use strict";
 
     var React = require("react"),
-        classnames = require("classnames");
+        classnames = require("classnames"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React),
+        ScrubbyMixin = require("js/jsx/mixin/Scrubby");
     
     var Label = React.createClass({
-        mixins: [React.addons.PureRenderMixin],
+        mixins: [FluxMixin, ScrubbyMixin],
+        
+        propTypes: {
+            size: React.PropTypes.string,
+            onScrub: React.PropTypes.func,
+            onScrubStart: React.PropTypes.func
+        },
+
+        /**
+         * inform the parent that scrubbing has started
+         *
+         * @private
+         */
+        _updatePositionBegin: function () {
+            this.props.onScrubStart();
+        },
+
+        /**
+         * Update the parents scrubby information
+         *
+         * @private
+         */
+        _updatePosition: function (clientX, clientY) {
+            var uiStore = this.getFlux().store("ui"),
+                rect = React.findDOMNode(this).getBoundingClientRect();
+
+            var value;
+            if (this.props.vertical) {
+                value = (rect.bottom - clientY) / rect.height;
+            } else {
+                value = (clientX - rect.left) / rect.width;
+            }
+
+            this.props.onScrub(uiStore.zoomCanvasToWindow(value));
```

Comment:
In fact, this entire function should bail at the beginning if `this.props.onScrub` is not defined. `getBoundingClientRect` is an expensive operation.